Library/Homebrew/extend/pathname.rb

```diff
@@ -376,7 +377,7 @@ def ensure_writable
     saved_perms = nil
     unless writable_real?
       saved_perms = stat.mode
-      chmod 0644
+      chmod "u+rw"
```

Comment:
Motivation for this change?

Reply:
As I recall, it related to modifying an executable script, and avoiding removing the executable bit.

Reply:
should I add a comment like `# preserve executable bit for executable scripts`

Reply:
?

Reply:
Executable scripts without shebangs are intentionally made non-executable unless `skip_clean` is used.

Reply:
I'm talking about executable scripts with a shebang line and that have the executable bit set. `chmod 0644` removes the executable bit, which is undesirable. `chmod "u+rw"` adds the read and write bits without modifying the executable bit. If it's set it remains set. If it's clear, it remains clear. This function is called `ensure_writable`.

Reply:
My mistake; I guess I'm trying to figure out why this is something that worked fine as-is on macOS but is causing Linuxbrew issues?

Reply:
The saved permissions are restored later
https://github.com/Homebrew/brew/pull/3377/files#diff-06540c12bb8e84922c0d329ed05b19bfR384
```ruby
chmod saved_perms if saved_perms
```
It's only transient that the script loses the executable bit.

Reply:
I'm still not comfortable running `chmod u+rw` in a method called _ensure\_**writable**_.  Either it should be `chmod u+w` or a different method name.

Reply:
The callers of this function read the file, modify it, and write it back to disk, so they need both read and write access. Perhaps it's a bit of a misnomer. I'm okay with that. Permission to write to a file usually implies permission to read a file, except in only the most unusual of exceptions.

Reply:
I could argue... but arguing without real-life examples, which I don't have handy, is useless, so, I'll leave it "as is" :)